modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js

```diff
@@ -506,32 +509,42 @@ const Options = ({
 		<div className="ddm-field-options-container">
 			<DragPreview component={Option}>{children}</DragPreview>
 
-			{fields.map((option, index) => (
-				<DnD
-					index={index}
-					key={option.id}
-					onDragEnd={composedMove}
-					option={option}
-				>
-					<Option disabled={disabled}>
-						{children({
-							defaultOptionRef,
-							expandedPanel: true,
-							fieldError,
-							handleBlur: composedBlur.bind(this, index),
-							handleField: !(fields.length - 1 === index)
-								? composedChange.bind(this, index)
-								: composedAdd.bind(this, index),
-							index,
-							onClick: () =>
-								handleConfirmDelete(index, option.value),
-							option,
-							showCloseButton:
-								!(fields.length - 1 === index) && !disabled,
-						})}
-					</Option>
-				</DnD>
-			))}
+			{fields.map((option, index) => {
+				return index !== fields.length - 1 ? (
+					<DnD
+						index={index}
+						key={option.id}
+						onDragEnd={composedMove}
+						option={option}
+					>
+						<Option disabled={disabled}>
+							{children({
+								defaultOptionRef,
+								expandedPanel: true,
+								fieldError,
+								handleBlur: composedBlur.bind(this, index),
+								handleField: composedChange.bind(this, index),
+								index,
+								onClick: () =>
+									handleConfirmDelete(index, option.value),
+								option,
+								showCloseButton:
+									!(fields.length - 1 === index) && !disabled,
+							})}
+						</Option>
+					</DnD>
+				) : null;
+			})}
+
+			<ClayButton
+				className="add-option-button"
+				displayType="secondary"
+				onClick={() => {
+					composedAdd.bind(this, fields.length - 1)('label', '');
```

Comment:
hey @cgmonte! not a change, i just wanted to understand how `composedAdd.bind(this, fields.length - 1)('label', '');` works, because i've never seen a function with two `( )` side by side like that

Reply:
When the user clicks the Add Option button, `composedAdd.bind()` returns a function that is [called immediately](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) with the passed parameters.

Reply:
I think that this may be simplified to `onClick={composedAdd.bind(this, fields.length - 1)}`. I added the `('label', '')` to ensure that it would start with an empty string as the Display Name. I cannot remember if I have faced a situation where it would fill it with something else.